### PR TITLE
`Forms`: Iterable collection of states

### PR DESCRIPTION
**Pull request by kaushikrw**
### Summary of changes

Depends on #254 to be merged.

- Provides a new collection type `StateCollection` that provides the `FormElement` and its `State`.
- The new type is iterable and immutable. A `MutableStateCollection` is provided that allows insertion of new elements.

### CODE REVIEW

**Review by sorenoid — CHANGES_REQUESTED** [review]
it all looks really good and works. I do think `StateCollection` needs to be more qualified as a name however even though it is internal.

File: toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/FeatureForm.kt [hunk 1]
```diff
@@ -157,7 +146,7 @@ internal fun FeatureFormContent(
 @Composable
 private fun FeatureFormBody(
     form: FeatureForm,
-    states: Map<Int, FormElementState>,
+    states: StateCollection,
```
[the remark this answers is not in the document]
Comment (kaushikrw) [reply toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/FeatureForm.kt]: `FormStateCollection` make sense. I have changed it.

**Review by sorenoid — APPROVED** [review]
nice work 👍

**Comment by kaushikrw** [comment]
thanks @sorenoid, merging!

**Pull request merged**